app/src/main/java/com/example/dinhh/soundscape/device/Library.kt

```diff
@@ -14,21 +18,17 @@ interface Library {
     fun beginSearch(selectedCategory: String): Completable
 }
 
-class LibraryImpl: Library{
+class LibraryImpl(private val sharedPref: SharedPref): Library{
     override fun beginSearch(selectedCategory: String): Completable {
         return Completable.create {
             Model.sounds.clear()
-            val apiKey =
-                "jFBaDxPcNzYZGu-gNMZ2L9-TjP1JjWl8OHFhdJV54gL82_M0cZi8oGEg-fB7gw3EpYvN0IHrHFP-Ic5sULo-iAWTl0k_y0t3CwrCQPpbYJkVIjmCV1Zzo0NB52ZLanwN"
+            val apiKey = sharedPref.getToken()
             try {
                 disposable =
                         wikiApiServe.getSounds(
-                            key = apiKey,
+                            key = apiKey.blockingGet(),
                             category = selectedCategory
-                        )
-                            .subscribeOn(Schedulers.io())
-                            .observeOn(AndroidSchedulers.mainThread())
-                            .subscribe(
+                        ).subscribe(
                                 { result ->
                                     for (i in result.indices) {
                                         Model.sounds.add(ListItem(result[i][0].Title))
```

Comment:
This code should go to data layer because it deals with the API. Device layer is kind of android thing (except for android local database) such as record, notification, camera, ... But we can do it together tomorrow